aptos-move/framework/aptos-stdlib/sources/data_structures/big_vector.move

```diff
@@ -131,14 +136,16 @@ module aptos_std::big_vector {
             vector::push_back(prev_bucket, t);
             cur_bucket_index = cur_bucket_index + 1;
         };
+        spec {
+            assert cur_bucket_index == num_buckets;
+        };
 
         // Shrink the table if the last vector is empty.
         let last_bucket = table_with_length::borrow_mut(&mut v.buckets, num_buckets - 1);
         if (vector::is_empty(last_bucket)) {
             move last_bucket;
             vector::destroy_empty(table_with_length::remove(&mut v.buckets, num_buckets - 1));
         };
-        v.end_index = v.end_index - 1;
```

Comment:
Why did this get deleted?

Reply:
This was moved above to line 123 to make the invariant more straightforward